core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -976,7 +975,11 @@ class WorkspaceService(
 
           // figure out which of the incoming aclUpdates are actually changes by removing all the existingAcls
           aclChanges = normalize(aclUpdates) -- existingAcls
-          _ = validateAclChanges(aclChanges, existingAcls, workspace)
+          callingUserRoles <- samDAO.listUserRolesForResource(SamResourceTypeNames.workspace,
```

Comment:
I wanted to just rely on what is in `existingAcls`, but if the calling user has the `projectOwner` role, they don't appear directly in `existingAcls` with their email address… instead there is a group for `projectOwner`, and presumably the calling user is a member of that group, but I don't know how to get that information.

This solution isn't ideal though because it does introduce another Sam call (though this method isn't called often, so that's a minor issue) and it breaks the pattern of going through the `workspaceAclManager` (though the Sam call could be hidden by adding a pass-through method to that interface, if folks felt like it was worth it).

Reply:
The additional sam call isn't ideal, but this does have the benefit of working in cases where the user has access to the workspace via group membership more generally than just with the project-owner case (e.g. a group was added as a writer on the workspace).

Reply:
We could add a new Sam API that returned a flattened policy membership for a resource if we wanted to, but we would still need the direct policy membership to calculate the ACL changes